objc/src/main/java/org/robovm/objc/ObjCClass.java

```diff
@@ -246,9 +247,8 @@ private static String getCustomClassName(Class<? extends ObjCObject> type) {
         String name = type.getName();
         if (customClassAnno != null && customClassAnno.value().length() > 0) {
             name = customClassAnno.value();
-        } else if (name.indexOf('.') == -1) {
-            name = "." + name;
         }
+        name = CUSTOM_CLASS_NAME_PREFIX + name.replace('.', '_');
```

Comment:
This should only be done if there is no `@CustomClass` annotation. If there is the name it specifies should be used as is and the user will have to make sure there are no collisions.